test/app.test.js

```diff
@@ -91,25 +93,95 @@ describe("app", function() {
+                                '1. Ward 82 (79800082), City of Johannesburg, Gauteng',
+                                '2. Ward 55 (79900055), City of Tshwane, Gauteng',
+                                '3. Back'
+                            ].join('\n')
+                        })
+                        .run();
+                });
+            });
+        });
+
+        describe("when the user submits a location that does not exist", function() {
+            it("should ask them how they would like to proceed", function() {
                 return tester
                     .setup.user.state('states:location')
-                    .input('Claremont')
+                    .input('Claremnt')
                     .check.interaction({
-                        state: 'states:results',
+                        state: 'states:incorrect_location',
                         reply: [
-                            'Select the location you would like to query:',
-                            '1. Ward 58 (19100058), City of Cape Town, Western Cape',
-                            '2. Ward 7 (52502007), Newcastle, KwaZulu-Natal',
-                            '3. Next',
-                            //'3. Ward 82 (79800082), City of Johannesburg, Gauteng',
-                            //'4. Ward 55 (79900055), City of Tshwane, Gauteng' 
-                        ].join('\n')
+                        'Location not found. Would you like to:',
+                        '1. Enter a different location',
+                        '2. Go to Main Menu',
+                        '3. Exit'
+                        ].join('\n')                       
                     })
                     .run();
             });
         });
 
+        describe("when a user enters an incorrect location, and then a valid location", function() {
+            describe("we want to show the results on multiple pages", function() {
+                it("should return page 1 of location results of the valid location", function() {
+                    return tester
+                        .setup.user.state('states:location')
+                        .inputs('Claremnt', '1', 'Claremont')
+                        .check.interaction({
+                            state: 'states:results',
+                                reply: [
+                                    'Select the location you would like to query:',
+                                    '1. Ward 58 (19100058), City of Cape Town, Western Cape',
+                                    '2. Ward 7 (52502007), Newcastle, KwaZulu-Natal',
+                                    '3. Next'
+                                ].join('\n')
+                            })
+                            .run();
+                    });
+
+                it("should page 2 of location results", function() {
+                    return tester
+                        .setup.user.state('states:location')
+                        .inputs('Claremont', '3')
```

Comment:
I don't think that this is quite testing what the description says it is testing. It should be `'Claremnt', '1', 'Claremont', '3'`, so that they first enter an incorrect location, and then a correct one.